crates/sovereign/sovereign-chain-components/src/sovereign/impls/sovereign_to_cosmos/client/create_client_payload.rs

```diff
@@ -28,59 +26,52 @@ use crate::sovereign::types::payloads::client::{
 */
 pub struct BuildSovereignCreateClientPayload;
 
-impl<Chain, Counterparty, DataChain> CreateClientPayloadBuilder<Chain, Counterparty>
+impl<Chain, Counterparty, Rollup, DataChain> CreateClientPayloadBuilder<Chain, Counterparty>
     for BuildSovereignCreateClientPayload
 where
     Chain: HasRollup<Rollup = Rollup>
         + HasDataChain<DataChain = DataChain>
         + HasCreateClientOptionsType<Counterparty, CreateClientOptions = SovereignCreateClientOptions>
         + HasCreateClientPayloadType<Counterparty, CreateClientPayload = SovereignCreateClientPayload>
-        + HasDataChain
-        + HasDataChainType<DataChain = DataChain>
-        + HasErrorType<Error = ReportError>, // TODO: Add chain dependencies for create client payload here
-    Chain::DataChain: HasErrorType + HasBlockingChainHandle,
+        + CanRaiseError<Rollup::Error>,
+    Rollup: CanQueryChainHeight<Height = RollupHeight>,
+    DataChain: HasBlockingChainHandle,
 {
     async fn build_create_client_payload(
         chain: &Chain,
         create_client_options: &SovereignCreateClientOptions,
     ) -> Result<SovereignCreateClientPayload, Chain::Error> {
-        // TODO: This will be replaced by data queried from the Roll-Up
+        // Build client state
+        let rollup_height = chain
```

Comment:
note that, `rollup_height` and `da_latest_height` must be from the same DA slot. If there was a delayer later, the DA may progress and give a future DA height - which is out of sync from `rollup_height`. So, it's best to use the `da_latest_height = rollup_height + rollup.genesis_da_height`.

Reply:
Since we are incrementing the height queried, https://github.com/informalsystems/hermes-sdk/blob/1abc19722024204398ae6ce24d41d6aef7ce36ce/crates/sovereign/sovereign-rollup-components/src/impls/queries/chain_status.rs#L32, adding the `rollup.genesis_da_height` would result in a height too high.
See 1abc19722024204398ae6ce24d41d6aef7ce36ce